File: CPP/sliding_window.cpp
#include <bits/stdc++.h>

using namespace std;
int maxsumsubarray(vector<int>& arr, int k) {
    if(arr.size() < k) return -1;
    int windowsum = 0;
    for(int i=0; i<k; i++) {
        windowsum += arr[i];
    }
    int maxsum = windowsum;
    for(int i=k; i<arr.size(); i++) {
        windowsum += (arr[i] - arr[i-k]);
        maxsum = max(maxsum, windowsum);
    }
    return maxsum;
}

File: PYTHON/sliding_window.py
def max_sum_subarray(arr, k):
    if len(arr) < k:
        return -1
    window_sum = 0
    for i in range(0, k):
        window_sum += arr[i]
    max_sum = window_sum
    for i in range(k, len(arr) ):
        window_sum += arr[i] - arr[i - k]
        max_sum = max(max_sum, window_sum)
    return max_sum
